Sources/Hypertext.swift

```diff
@@ -6,18 +6,23 @@
 //  Copyright © 2016 Sahand Nayebaziz. All rights reserved.
 //
 
-public protocol Renderable {
-    func render() -> String
-    func render(startingWithSpacesCount: Int) -> String
```

Comment:
My goals are to:
- make the autocomplete that pops up on the end of a `Renderable` show only `render()` and our second `render(indent/startingWithSpacesCount)`
- set `Renderable` up so that making a custom type conform to `Renderable` means you implement `render()` which is the basic, nothing-added output of your instance to HTML, and then also implement `render(indent/startingWithSpacesCount)` so that you specifically consider the case where formatting is happening. 

For example, if somebody has a tag that is some sort of document comment, maybe they always want that to be at the very beginning of a line all the way to the side with no indentation, so in the formatting `render` they can specifically ignore any spacing/indentation. Or if they have some type that's sort of like an array that carries items in it, they can iterate through their items in the formatting `render` and add their own nested indents.